internal/rollup/rollup.bzl

```diff
@@ -0,0 +1,154 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Rules for production rollup bundling.
+"""
+load("//internal:node.bzl", nodejs_binary = "nodejs_binary_macro")
+load("//internal:common/module_mappings.bzl", "module_mappings_runtime_aspect")
+load("//internal:collect_es6_sources.bzl", "collect_es6_sources")
+
+def _es6_consumer(ctx):
+  sources = collect_es6_sources(ctx)
+  
+  return [DefaultInfo(
+      files = sources,
+      runfiles = ctx.runfiles(sources.to_list()),
+  )]
+
+es6 = rule(
+    implementation = _es6_consumer,
+    attrs = {
+        "deps": attr.label_list()
+    }
+)
+
+def _sources_aspect_impl(target, ctx):
+  result = depset()
+  if hasattr(ctx.rule.attr, "deps"):
+    for dep in ctx.rule.attr.deps:
+      if hasattr(dep, "node_sources"):
+        result += dep.node_sources
+  # Note layering: until we have JS interop providers, this needs to know how to
+  # get TypeScript outputs.
+  if hasattr(target, "typescript"):
+    result += target.typescript.es5_sources
+  elif hasattr(target, "files"):
+    result += target.files
+  return struct(node_sources = result)
+
```

Comment:
Ah interesting. What does `_sources_aspect_impl` do here? I am not sure I follow this part.